src/examples/AddDataExample.java

```diff
@@ -0,0 +1,194 @@
+    // function.
+    try {
+      byteMetricUID = tsdb.getUID(UniqueIdType.METRIC, metricName);
+    } catch (IllegalArgumentException iae) {
+      System.out.println("Metric name not valid.");
+      iae.printStackTrace();
+    } catch (NoSuchUniqueName nsune) {
+      // If not, great. Create it.
+      byteMetricUID = tsdb.assignUid("metric", metricName);
+    }
+
+    // Make a single datum
+    long timestamp = System.currentTimeMillis();
+    long value = 314159;
+    // Make key-val
+    Map<String, String> tags = new HashMap<String, String>(1);
+    tags.put("dummy-key", "dummy-val1");
+
+    // This is an optional errorback to handle when there is a failure.
+    class errBack implements Callback<String, Exception> {
+      public String call(final Exception e) throws Exception {
+        String message = ">>>>>>>>>>>Failure!>>>>>>>>>>>";
+        System.out.println(message);
+        return message;
+      }
+    }
+    ;
+
+    // Start timer
+    long startTime1 = System.nanoTime();
+    // Write datum to tsdb
+    Deferred<Object> deferred = tsdb.addPoint(metricName, timestamp, value,
+        tags);
+
+    // Add the callbacks to the deferred object. (They might have already
+    // returned, btw)
+    // This will cause the calling thread to wait until the add has completed.
+    deferred.addErrback(new errBack());
+
+    // End timer.
+    long elapsedTime1 = (System.nanoTime() - startTime1) / (1000 * 1000);
+    System.out.println("\nAdding 1 point took: " + elapsedTime1
+        + " milliseconds.\n");
+
+    /*
+     * Now let's input a stream of data
+     */
+
+    // create an array for storing the deferreds of each addPoint() call
+    final int nDataPoints = 100;
+    final ArrayList<Deferred<Object>> deferreds = new ArrayList<Deferred<Object>>(
+        nDataPoints);
+    // Initialize which time series you're going to write to
+    WritableDataPoints wdp = tsdb.newDataPoints();
+    wdp.setSeries(metricName, tags); // you can recall this at any time to
+                                     // change the ts.
+    // Make some data
+    Random rand = new Random();
+    long[] timestamps1 = makeTimestamps(nDataPoints, rand);
+    long[] values1 = makeRandomValues(nDataPoints, rand);
+    // Start timer
+    long startTime2 = System.nanoTime();
+    // Now write to tsdb
+    for (int i = 0; i < nDataPoints; i++) {
+      // write the data and add the returned Deferred to our ArrayList
+      deferreds.add(wdp.addPoint(timestamps1[i], values1[i]));
+    }
+
+    try {
+      Deferred.groupInOrder(deferreds).addErrback(new errBack())
+          .joinUninterruptibly();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    // End timer.
+    long elapsedTime2 = (System.nanoTime() - startTime2) / (1000 * 1000);
+    System.out.println("\nAdding " + nDataPoints
+        + " points individually took: " + elapsedTime2 + " milliseconds.\n");
+
+    /*
+     * Now try writing in batch import mode
```

Comment:
Some more comments about the difference between batched and straight mode would be helpful here in the example.